bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/BLockStore.java

```diff
@@ -40,9 +40,18 @@ public void addLockToMap(String lockName) {
         globalLockMap.put(lockName, new BLock());
     }
 
-    public synchronized BLock getLockFromMap(String lockName) {
+    public BLock getLockFromMap(String lockName) {
         return globalLockMap.computeIfAbsent(lockName, (k) -> {
             return new BLock();
         });
     }
+
+    public void panicIfInLock(String lockName) {
+        BLock lock = globalLockMap.get(lockName);
+        if (lock != null) {
+            if (!lock.isLockFree()) {
```

Comment:
We can combine the checks right?

```suggestion
        if (lock != null && !lock.isLockFree()) {
```